lib/publiccloud/azure.pm

```diff
@@ -24,6 +24,12 @@ has storage_account => 'openqa';
 has container       => 'sle-images';
 has lease_id        => undef;
 
+sub decode_json_trim {
```

Comment:
@jlausuch , @cfconrad  looking for better name . any ideas ?

Reply:
What about a method: `run_az()` do these special things inside this method and return json?

Reply:
`decode_azure_json`

Reply:
@cfconrad - I like your idea but I think it would better go in separate PR , since it requires some refactoring which is out of scope for this PR . 
@jlausuch - for now I will go with yours and later we can implement Clemens' idea